libbpf-tools/vfsstat.c

```diff
@@ -156,13 +156,46 @@ int main(int argc, char **argv)
 		return 1;
 	}
 
-	obj = vfsstat_bpf__open_and_load();
-	if (!obj) {
-		fprintf(stderr, "failed to open and/or load BPF object\n");
+	skel = vfsstat_bpf__open();
+	if (!skel) {
+		fprintf(stderr, "failed to open BPF skelect\n");
 		return 1;
 	}
 
-	err = vfsstat_bpf__attach(obj);
+	/* It fallbacks to kprobes when kernel does not support fentry. */
+	if (fentry_exists("vfs_read", NULL)) {
```

Comment:
If fentry for vfs_read is supported ("exists"), there is no need to repeat the check for other vfs_xxx operations. Please roll everything under single if/else check.

Reply:
Fixed it